test/run.sh

```diff
@@ -47,13 +49,51 @@ if [ $# -eq 0 ]; then
 	exit 1
 fi
 
-# load config lists
-# contains:
-#   globalTests
-#   testAlias
-#   imageTests
-#   globalExcludeTests
-. "$dir/config.sh"
+# declare configuration variables
+declare -a globalTests=()
+declare -A testAlias=()
+declare -A imageTests=()
+declare -A globalExcludeTests=()
+
+# if there are no user-specified configs, use the default config
+if [ ${#configs} -eq 0 ]; then
+	configs+=("$dir/config.sh")
+fi
+
+# load the configs
+declare -A testPaths=()
+for conf in "${configs[@]}"; do
+	. "$conf"
+
+	# Determine the full path to any newly-declared tests
+	confDir="$(dirname "$conf")"
+
+	for testName in ${globalTests[@]}; do
+		[ "${testPaths[$testName]}" ] && continue
+
+		if [ -d "$confDir/tests/$testName" ]; then
+			# Test directory found relative to the conf file
+			testPaths[$testName]="$confDir/tests/$testName"
+		elif [ -d "$dir/tests/$testName" ]; then
+			# Test directory found in the main tests/ directory
+			testPaths[$testName]="$dir/tests/$testName"
+		fi
```

Comment:
Specifically here: https://github.com/appropriate/docker-library-official-images/blob/additional-test-config/test/run.sh#L154

Although now that I look closer, it seems like `$scriptDir` will be blank in that case and result in a nonsensical error message.

Where would prefer it be caught? I'm thinking having an `elif [ -z "$scriptDir" ]` check after the `if [ -d "$scriptDir" ]` check might be good. Either that or just change the error message in the `else` block from `echo >&2 "error: $scriptDir is not a directory"` to `echo >&2 "error: unable to locate test '$t'"`